### PR TITLE
Bigtable: cleanup unused variable and add deprecation warnings for to be removed features

### LINKED ISSUE #7531: Bigtable: cleanup and warnings about support for backwards compatible features that will be discontinued.
Remove unused [variable](https://github.com/googleapis/google-cloud-python/blob/master/bigtable/google/cloud/bigtable/instance.py#L33).

Add `DeprecationWarnings` for backwards compatible features that should/will go away.

- `table.yield_rows()`
- `location_id`, `serve_nodes` and `default_storage_type` parameters in `instance.create()`

### PR DESCRIPTION
**Pull request by AVaksman**
Fixes #7531 
* some cleanup
* add deprecation warnings for backwards compatible features that will go away

### CODE REVIEW

File: bigtable/google/cloud/bigtable/instance.py [hunk 1]
```diff
@@ -276,6 +282,14 @@ def create(
         """
 
         if clusters is None:
+            warnings.warn(
+                _INSTANCE_CREATE_WARNING.format(
+                    "location_id", "serve_nodes", "default_storage_type"
+                ),
+                PendingDeprecationWarning,
```
Comment (tseaver) [inline bigtable/google/cloud/bigtable/instance.py]: I think we can make this just a regular `DeprecationWarning`.  Thanks for adding `stacklevel=2`!

**Comment by tseaver** [comment]
@AVaksman We need to update the unit tests to catch / assert the new warnings being raised, e.g.:

```python
    def test_foo(self):
        import warnings

        with warnings.catch_warnings(record=True) as warned:
            foo()

        self.assertEqual(len(warned), 1)
        self.assertTrue(issubclass(warned[0].category, DeprecationWarning)
```

File: bigtable/google/cloud/bigtable/instance.py (hunk 1 above)
Comment (AVaksman) [reply bigtable/google/cloud/bigtable/instance.py]: Done, thanks

**Comment by AVaksman** [comment]
> @AVaksman We need to update the unit tests to catch / assert the new warnings being raised, e.g.:
> 
> ```python
>     def test_foo(self):
>         import warnings
> 
>         with warnings.catch_warnings(record=True) as warned:
>             foo()
> 
>         self.assertEqual(len(warned), 1)
>         self.assertTrue(issubclass(warned[0].category, DeprecationWarning)
> ```

Done, thanks

**Comment by tseaver** [comment]
Unrelated test failures:

- The [`api_core` failure](https://source.cloud.google.com/results/invocations/8740f11f-5c76-40ca-a874-756d65817ca4/targets/cloud-devrel%2Fclient-libraries%2Fgoogle-cloud-python%2Fpresubmit%2Fapi_core/log) seems clearly unrelated to this PR:  it fails in the `pytype` job.
- The [`storage` failure](https://source.cloud.google.com/results/invocations/47008c5f-f662-4544-a8b1-85ed9798b4f0/targets/cloud-devrel%2Fclient-libraries%2Fgoogle-cloud-python%2Fpresubmit%2Fstorage/log) is a known flaky system test, #7505.

**Pull request merged**